src/safeposix/syscalls/net_calls.rs

```diff
@@ -36,29 +41,45 @@ impl Cage {
         return sockfd;
     }
 
+    //Find an available file descriptor index in the File Descriptor Table
+    //If available, insert the socket fd into the File Descriptor Table at the 
+    //available index
     fn _socket_inserter(&self, sockfd: FileDescriptor) -> i32 {
         let (fd, guardopt) = self.get_next_fd(None);
+        //Indicate an error in the fd and propogate the error forward 
         if fd < 0 {
             return fd;
         }
         let fdoption = &mut *guardopt.unwrap();
-        let _insertval = fdoption.insert(sockfd);
+        //Insert the sockfd as the FileDescriptor inside fdoption
```

Comment:
Why?  What are we doing here?  

(We can assume the reader knows Rust well enough to know they are inserting  a sockfd inside the fdoption datastructure)